apple/swift-docc
# changes to have diagnostics only for known directives

## Summary

Currently there are confusing "Directives are not supported in symbol source documentation" while building Doxgyen content. 


## Dependencies
No known dependencies, 

## Testing

Steps
1. Build docc locally with these changes
2. Create a new project in Xcode and an Objective class to it.3. 
3. In a Objective-C class, add a method and document it with Doxygen tags4. 
 /// Brief description of this method
 @Image something Description of this Image
 @params Description of this parameter 
 @returns Description of return value
- (NSString *)methodNameWithSomething:(NSString *)something;
4. Build documentation for the target by pointing docc_exec to the locally build docc. 
5. There shouldn’t be any warnings for the @param and @returns lines and there should be one for @Image. @Image is a known directive and hence we expect only that one to have warning. 
## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ 👍🏻] Added tests
- [ 👍🏻] Ran the `./bin/test` script and it succeeded
- [ 👍🏻] Updated documentation if necessary
